src/algorithms/WState.cpp

```diff
@@ -0,0 +1,26 @@
+#include "algorithms/WState.hpp"
+
+namespace qc {
+void fGate(QuantumComputation& qc, std::size_t i, std::size_t j, std::size_t n,
+           std::size_t k) {
+  auto theta = std::acos(std::sqrt(1.0 / (n - k + 1)));
```

Comment:
```suggestion
  const auto theta = std::acos(std::sqrt(1.0 / static_cast<double>(n - k + 1)));
```
Should get rid of the clang-tidy warning.